### PR TITLE
add mrevenge with RND

### PR DESCRIPTION
**Pull request by hnyu**
Update `rnd_algorithm.py` and add the example of training MRevenge.

### CODE REVIEW

File: alf/utils/common.py [hunk 1]
```diff
@@ -1078,3 +1078,18 @@ def set_random_seed(seed, eager_mode):
     random.seed(seed)
     np.random.seed(seed)
     tf.random.set_seed(seed)
+
+
+@gin.configurable
```
Comment (emailweixu) [inline alf/utils/common.py]: TensorSpec is a gin configurable defined in utils/external_configurables.py

File: alf/algorithms/rnd_algorithm.py [hunk 2]
```diff
@@ -91,29 +124,38 @@ def train_step(self,
...
             observation = self._observation_normalizer.normalize(observation)
 
+        if self._encoder_net is not None:
+            observation = tf.stop_gradient(self._encoder_net(observation)[0])
+
         pred_embedding, _ = self._predictor_net(observation)
         target_embedding, _ = self._target_net(observation)
 
-        loss = 0.5 * tf.reduce_mean(
+        loss = tf.reduce_sum(
```
Comment (emailweixu) [inline alf/algorithms/rnd_algorithm.py]: why change to reduce_sum?

File: alf/examples/ppo_rnd_mrevenge.gin [hunk 3]
```diff
@@ -0,0 +1,78 @@
+# RND on MontezumaRevenge
```
Comment (emailweixu) [inline alf/examples/ppo_rnd_mrevenge.gin]: add to train_play_test.py

File: alf/algorithms/rnd_algorithm.py [hunk 4]
```diff
@@ -44,30 +43,64 @@ class RNDAlgorithm(Algorithm):
...
+                 reward_adapt_speed=None,
                  observation_adapt_speed=None,
                  observation_spec=None,
+                 learning_rate=None,
+                 clip_value=-1.0,
+                 stacked_frames=True,
                  name="RNDAlgorithm"):
         """
         Args:
+            encoder_net (Network): a shared network that encodes imgs to
```
Comment (Haichao-Zhang) [inline alf/algorithms/rnd_algorithm.py]: ```encodes imgs ...```, is there an explicit assumption that the inputs have to be images? It seems not.

Comment (hnyu) [reply alf/algorithms/rnd_algorithm.py]: If it's not, then this can be None

File: alf/algorithms/rnd_algorithm.py (hunk 2 above)
Comment (hnyu) [reply alf/algorithms/rnd_algorithm.py]: That's what I usually have for experiments

Comment (Haichao-Zhang) [reply alf/algorithms/rnd_algorithm.py]: Is this related to why a small learning rate can be used for training with good results in your experiments?

Comment (hnyu) [reply alf/algorithms/rnd_algorithm.py]: I never used a small learning rate? This is not learning improvements, just RND

File: alf/examples/rnd_super_mario.gin [hunk 5]
```diff
@@ -65,6 +67,7 @@ TrainerConfig.use_tf_functions=True
 TrainerConfig.debug_summaries=True
 TrainerConfig.summarize_grads_and_vars=False
 TrainerConfig.summary_interval=10
+TrainerConfig.update_counter_every_mini_batch=True
```
Comment (emailweixu) [inline alf/examples/rnd_super_mario.gin]: Is this just for debug?

Comment (hnyu) [reply alf/examples/rnd_super_mario.gin]: For updating the curve more frequently, which is what I always use

Comment (emailweixu) [reply alf/examples/rnd_super_mario.gin]: But this does not have effect for on_policy_trainer.

Comment (hnyu) [reply alf/examples/rnd_super_mario.gin]: I thought this example uses PPO
